src/main/java/com/dreamsportslabs/guardian/service/OidcTokenService.java

```diff
@@ -176,7 +177,12 @@ private Single<OidcTokenResponseDto> clientCredentialsFlow(
   }
 
   private Single<OidcTokenResponseDto> refreshTokenFlow(
-      TokenRequestDto requestDto, String tenantId, String authorizationHeader) {
+      TokenRequestDto requestDto,
+      String tenantId,
+      String authorizationHeader,
+      MultivaluedMap<String, String> headers) {
+    TenantConfig tenantConfig = registry.get(tenantId, TenantConfig.class);
+    TokenConfig tokenConfig = tenantConfig.getTokenConfig();
     return authenticateClient(requestDto, tenantId, authorizationHeader)
```

Comment:
side note, if the client is public, then there is no way to authenticate client. 
Let's check the RFC once to see if client is public, then is client-id and client-secret required to validate the identity.

Reply:
same applies to the token endpoint as well. 
Client id and secret validation not required if PKCE is mandatory for all clients, including public ones.